modules/core/src/smithy4s/schema/DefaultValueSchemaVisitor.scala

```diff
@@ -18,128 +18,79 @@ package smithy4s
-            Some(Document.fromString("1970-01-01T00:00:00.00Z"))
-          case TimestampFormat.HTTP_DATE =>
-            Some(Document.fromString("Thu, 01 Jan 1970 00:00:00 GMT"))
-          case TimestampFormat.EPOCH_SECONDS => Some(Document.fromLong(0L))
-        }
-      case PBlob       => Some(Document.fromString(""))
-      case PBigInt     => Some(Document.fromBigDecimal(BigDecimal(0)))
+      case PShort      => Some(0: Short)
+      case PString     => Some("")
+      case PFloat      => Some(0f)
+      case PDouble     => Some(0d)
+      case PInt        => Some(0)
+      case PLong       => Some(0L)
+      case PBoolean    => Some(false)
+      case PTimestamp  => Some(Timestamp.epoch)
+      case PBlob       => Some(ByteArray.empty)
+      case PBigInt     => Some(BigInt(0))
+      case PBigDecimal => Some(BigDecimal(0))
+      case PDocument   => Some(Document.DNull)
       case PUUID       => None
-      case PInt        => Some(Document.fromInt(0))
-      case PBigDecimal => Some(Document.fromBigDecimal(BigDecimal(0)))
-      case PBoolean    => Some(Document.fromBoolean(false))
-      case PLong       => Some(Document.fromLong(0))
       case PByte       => None
       case PUnit       => None
-      case PDocument   => Some(Document.DNull)
     }
-  )
 
   def collection[C[_], A](
       shapeId: ShapeId,
       hints: Hints,
       tag: CollectionTag[C],
       member: Schema[A]
-  ): GetDefaultValue[C[A]] = from(hints, Some(Document.array()))
+  ): Option[C[A]] = Some(tag.empty)
 
   def map[K, V](
       shapeId: ShapeId,
       hints: Hints,
       key: Schema[K],
       value: Schema[V]
-  ): GetDefaultValue[Map[K, V]] = from(hints, Some(Document.obj()))
+  ): Option[Map[K, V]] = Some(Map.empty)
 
   def enumeration[E](
       shapeId: ShapeId,
       hints: Hints,
       values: List[EnumValue[E]],
       total: E => EnumValue[E]
-  ): GetDefaultValue[E] = from(hints, None)
+  ): Option[E] = None
 
   def struct[S](
       shapeId: ShapeId,
       hints: Hints,
       fields: Vector[SchemaField[S, _]],
       make: IndexedSeq[Any] => S
-  ): GetDefaultValue[S] = from(hints, None)
+  ): Option[S] = None
 
   def union[U](
       shapeId: ShapeId,
       hints: Hints,
       alternatives: Vector[SchemaAlt[U, _]],
       dispatch: Alt.Dispatcher[Schema, U]
-  ): GetDefaultValue[U] = from(hints, None)
+  ): Option[U] = None
 
   def biject[A, B](
       schema: Schema[A],
       bijection: Bijection[A, B]
-  ): GetDefaultValue[B] = {
-    val res = apply(schema)
-    new GetDefaultValue[B] {
-      def getDefault: Option[Document] = res.getDefault
-    }
+  ): Option[B] = {
+    schema.compile(this).map(bijection.to)
   }
 
   def refine[A, B](
       schema: Schema[A],
       refinement: Refinement[A, B]
-  ): GetDefaultValue[B] = {
-    val res = apply(schema)
-    new GetDefaultValue[B] {
-      def getDefault: Option[Document] = res.getDefault
-    }
-  }
+  ): Option[B] = None
```

Comment:
At first, I was hopeful this would solve: https://github.com/disneystreaming/smithy4s/issues/791, but because the `SmithyToIR` works directly with Smithy `Shape`, I don't think it will... right?

Relevant decision happens in this piece of code: https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala#L845

Reply:
You are correct, it won't solve that issue, unfortunately